### DISCUSSION TITLE
Preserve LHS quotes when converting to properties style

### DISCUSSION THREAD

**Discussion by geekosphere-net** [mikefarah/yq#818] [Q&A] [answered]
Saw [discussion 699](https://github.com/mikefarah/yq/discussions/699) when trying to migrate from v3 to v4 where we used `--printMode pv` to "flatten" or convert our yaml to a property string, and while it works for most of our files we just ran into a case where we have a value in the YAML that is quoted and we lose the quotes, any idea how to preserve them?

YAML:
```
one:
  "two.three":
    four: "five"
```
when we use the suggestion in [699](https://github.com/mikefarah/yq/discussions/699) we get
```
one.two.three.four=five
```
but what we would like to see is
```
one."two.three".four=five
```
Is that possible? Not all that familiar with `jq`. Tried looking at using `style="double"` or `--unwrapScalar=false` but neither seemed to help.
Thanks in advance

**Comment by mikefarah** [comment, marked as answer]
The quotes on that key are part of the yaml formatting style as opposed to the value - so they do not come through when processing the data.

The closest I can get is to wrap all the properties:

```
yq eval '.. | select((tag == "!!map" or tag == "!!seq") | not) | (path | .[] |= sub("(.*)", "\"${1}\"") | join(".")) + "=" + .' examples/data1.yaml
```

to get:

```
"one"."two.three"."four"=five
```

Not sure if that helps? 

Note that this requires v4.9.1 to be able to escape double quotes in a double quoted string.

**Reply by mikefarah** [reply]
It may be possible to come up with an expression that only wraps strings with `.` in their value by counting how many elements in split('.') there are - that expression would start getting pretty hairy though I think

**Reply by geekosphere-net** [reply]
Thanks for that query I don't think I would have ever figured that out, and having more strings/quotes is better than not having them.
As we are using this as part of a CI/CD process I might be able to use `awk` or something to determine which strings have a `.` in them and need special attention.

Thanks!
